src/System.Private.CoreLib/src/System/Threading/Thread.CoreRT.Windows.cs

```diff
@@ -32,6 +32,11 @@ private void PlatformSpecificInitialize()
         {
         }
 
+        private static void PlatformSpecificGlobalInitialize()
+        {
+            RuntimeImports.RhSetThreadExitCallback(AddrofIntrinsics.AddrOf<Action>(OnThreadExit));
```

Comment:
I've introduced this for both Win and Unix. On Unix I've moved callback registration from ctor to cctor since I found no reason to register same callback on every thread creation.

Reply:
static constructor will make add overhead of cctor check to every static method in the type. It is better to register the callback on every thread creation.

Reply:
Ah I remember such discussions on CoreCLR repo back in the days it was opensourced. AFAIK there is a trick to not dodge penalty by doing something like
```csharp

private static s_not_used = PlatformSpecificGlobalInitialize();

```

Reply:
But then you have to access the variable to trigger the initialization.

Reply:
The typical pattern in CoreLib is to lazy initialize, for example like this: https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/Environment.cs#L152

> if you think that we are fine with registration in instance ctor and alternatives would not provide noticeable benefits.

I think so.

Reply:
Did I get it correct, that you suggesting to introduce static bool filed indicating that we registered the callback and conditionally register it in instance ctor? If that is not what you mean, how do you find it?

Reply:
BTW there are 82 issues in runtime repo mentioning `beforefieldinit`. Quite a hot topic.

Reply:
I would keep it as it was before. There is no harm in setting the callback multiple times.